### PR TITLE
Removing support for python 3.12. ORM only supports up to 3.11

### PR DESCRIPTION
**Pull request by calebevans**
Support for Python 3.12 was removed because the [ORM only supports up to 3.11](https://blog.ponyorm.org/2023/09/26/pony-orm-release-0-7-17/). Using 3.12 seemed to cause some issues with using `lambda`.

### CODE REVIEW

**Review by myakove — APPROVED** [review]

**Review by myakove — CHANGES_REQUESTED** [review]
Can you please add comment to the issue that add support for 3.12 for easy tracking?

**Comment by calebevans** [comment]
> Can you please add comment to the issue that add support for 3.12 for easy tracking?

@myakove Is this what you're looking for? https://github.com/RedHatQE/qe-metrics/issues/66#issuecomment-2028098165

File: pyproject.toml [hunk 1]
```diff
@@ -35,7 +35,7 @@ Download = "https://github.com/RedHatQE/qe-metrics"
 "Bug Tracker" = "https://github.com/RedHatQE/qe-metrics/issues"
 
 [tool.poetry.dependencies]
-python = "^3.8"
+python = ">=3.8,<3.12"
```
Comment (myakove) [inline pyproject.toml:38]: I mean add here comment with https://github.com/ponyorm/pony/issues/703

**Review by myakove — CHANGES_REQUESTED** [review]
.

**Review by myakove — APPROVED** [review]

**Pull request merged**